tck/src/main/java/ee/jakarta/tck/data/web/validation/PersistenceTests.java

```diff
@@ -0,0 +1,238 @@
+                new Rectangle("RECT-012", 1801L, 5L, 5, 5), //Max x
+                new Rectangle("RECT-013", 5L, 1001L, 5, 5), //Max y
+                new Rectangle("RECT-014", 5L, 5L, 121, 5), //Max width
+                new Rectangle("RECT-014", 5L, 5L, 5, 81) //Max height
+                );
+        
+        resultingException = assertThrows(ConstraintViolationException.class, () -> {
+            rectangles.saveAll(invalidRects);
+        });
+        assertEquals(0, rectangles.count(), "No rectangles should have presisted to database while violating constraints.");
+        assertEquals(5, resultingException.getConstraintViolations().size(), "Incorrect number of constraint violations.");        
+    }
+    
+    @Assertion(id = "133", strategy="Ensure that entities that do not violate the constraints can be successfully updated in the database")
+    public void testUpdateWithValidConstraints() {
+        List<Rectangle> resultRects;
+        
+        // Save
+        Rectangle validRect = new Rectangle("RECT-020", 5L, 5L, 5, 5);
+        rectangles.save(validRect);
+        
+        assertEquals(1, rectangles.count(), "Number of results was incorrect");
+        
+        // Update
+        Rectangle updatedRect = new Rectangle("RECT-020", 10L, 10L, 10, 10);
+        rectangles.save(updatedRect);
+        
+        assertEquals(1, rectangles.count(), "Number of results was incorrect");
+        
+        // Verify
+        resultRects = getResults();
+        Rectangle resultRect = resultRects.get(0);
+        assertRectangleFields(updatedRect, resultRect);      
+    }
+    
+    @Assertion(id = "133", strategy="Ensure that entities that do not violate the constraints can be successfully updated in the database")
+    public void testUpdateAllWithValidConstraints() {
+        List<Rectangle> resultRects;
+        
+        // Save All
+        List<Rectangle> validRects = List.of(
+                new Rectangle("RECT-030", 5L, 5L, 5, 5),
+                new Rectangle("RECT-031", 0L, 0L, 1, 1),
+                new Rectangle("RECT-032", 1800L, 1000L, 120, 80),
+                new Rectangle("RECT-033", 1L, 1L, 1, 1)
+                );
+        rectangles.saveAll(validRects);
+        assertEquals(4, rectangles.count(), "Number of results was incorrect");
+        
+        // Update All
+        List<Rectangle> updatedRects = List.of(
+                new Rectangle("RECT-030", 6L, 5L, 5, 5),
+                new Rectangle("RECT-031", 0L, 1L, 1, 1),
+                new Rectangle("RECT-032", 1800L, 999L, 120, 80),
+                new Rectangle("RECT-033", 1L, 1L, 1, 2)
+                );
+        rectangles.saveAll(updatedRects);
+        assertEquals(4, rectangles.count(), "Number of results was incorrect");
+        
+        // Verify All
+        resultRects = getResults();
+        for(int i = 0; i < resultRects.size(); i++) {
+            assertRectangleFields(updatedRects.get(i), resultRects.get(i));
```

Comment:
Similar to above comment:
```suggestion
        updatedRects = rectangles.saveAll(updatedRects);
        assertEquals(4, rectangles.count(), "Number of results was incorrect");
        
        // Verify All
        resultRects = getResults();
        for(int i = 0; i < resultRects.size(); i++) {
            assertRectangleFields(updatedRects.get(i), resultRects.get(i));
```